### PR TITLE
Lint test files: Files outside of subdirectories, `circuit_graph`

### PR DESCRIPTION
**Pull request by dwierichs**
More linting of tests.

### CODE REVIEW

File: tests/test_qubit_device.py [hunk 1]
```diff
@@ -343,7 +346,7 @@ def test_results_created(self, mock_qubit_device_extract_stats, monkeypatch, mea
 
         qscript = QuantumScript(measurements=[measurement])
 
-        with monkeypatch.context() as m:
+        with monkeypatch.context():
```
Comment (timmysilv) [inline tests/test_qubit_device.py:349]: is this needed to make the mock device work? same question in general for when the context variable is not used (and now removed) throughout the PR

**Review by timmysilv — APPROVED** [review]
thanks for doing this! in case you didn't know, you can put a pylint disable comment at a higher context, and then it's disabled for everything within that context (eg. top of the file disables that warning for the whole file), and we don't mind if it's at a high scope (within reason)

File: tests/test_qubit_device.py (hunk 1 above)
Comment (dwierichs) [reply tests/test_qubit_device.py:349]: ...good question :sweat_smile:

File: tests/test_qnode.py [hunk 2]
```diff
@@ -50,8 +51,7 @@ def test_changing_invalid_interface(self):
         test_interface = "something"
 
         @qnode(dev)
-        def circuit(x):
-            qml.RX(wires=0)
```
[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/test_qnode.py:54]: Can't raise an error on an invalid operation instantiation if you never get to call the function :D

File: tests/test_qubit_device.py (hunk 1 above)
Comment (dwierichs) [reply tests/test_qubit_device.py:349]: The answer is no :) I think this was just copy-paste legacy, and the mocking is actually happening in the fixtures, so no need to open the context in the test function itself.

File: tests/tests_passing_pylint.txt [hunk 3]
```diff
@@ -35,13 +36,33 @@ tests/qchem/test_givens_rotations.py
 tests/tape/test_qscript.py
```
Comment (timmysilv) [inline tests/tests_passing_pylint.txt:36]:
```suggestion
tests/qnn/**
tests/tape/test_qscript.py
```

**Review by timmysilv — APPROVED** [review]

File: tests/circuit_graph/test_circuit_graph.py [hunk 4]
```diff
@@ -43,32 +42,31 @@ def ops():
...
+@pytest.fixture(name="circuit")
+def circuit_fixture(ops, obs):
+    """A fixture of a circuit generated based on the ops and obs fixtures above."""
+    return CircuitGraph(ops, obs, Wires([0, 1, 2]))
 
 
-@pytest.fixture
-def parameterized_circuit_gaussian(wires):
+@pytest.fixture(name="parameterized_circuit_gaussian")
+def parameterized_circuit_gaussian_name(wires):
```
Comment (Jaybsoni) [inline tests/circuit_graph/test_circuit_graph.py:61]: Is this supposed to be fixture like the rest? 
```suggestion
def parameterized_circuit_gaussian_fixture(wires):
```

File: tests/test_device.py [hunk 5]
```diff
@@ -167,7 +167,7 @@ def get_device(wires=1):
 
 
 @pytest.fixture(scope="function")
-def mock_device_arbitrary_wires(monkeypatch, wires):
+def mock_device_arbitrary_wires(monkeypatch):
```
Comment (Jaybsoni) [inline tests/test_device.py:170]: Should we rename this with `fixture` and add a name to the decorator as with the other fixtures?

File: tests/test_device.py [hunk 6]
```diff
@@ -436,19 +437,19 @@ def circuit_gauss(mag_alpha, phase_alpha, phi):
...
+        (1, Wires([0])),
+        (4, Wires([1, 3])),
+        (["a", 2], Wires([2])),
+        (["a", 2], Wires([2, "a"])),
     ]
 
-    @pytest.mark.parametrize("dev_wires, wires_to_map, res", wires_to_try)
-    def test_map_wires_caches(self, dev_wires, wires_to_map, res, mock_device):
+    @pytest.mark.parametrize("dev_wires, wires_to_map", wires_to_try)
+    def test_map_wires_caches(self, dev_wires, wires_to_map, mock_device):
```
Comment (Jaybsoni) [inline tests/test_device.py:452]: was the last argument being ignored here?

**Review by Jaybsoni — COMMENTED** [review]
Added some comments!

File: tests/circuit_graph/test_circuit_graph.py (hunk 4 above)
Comment (dwierichs) [reply tests/circuit_graph/test_circuit_graph.py:61]: Ah, yes!

File: tests/test_device.py (hunk 6 above)
Comment (dwierichs) [reply tests/test_device.py:452]: Exactly, it simply is not used

File: tests/test_device.py (hunk 5 above)
Comment (dwierichs) [reply tests/test_device.py:170]: I'm not sure I follow, sorry. `lint` does not complain about the name being redefined, which it does for fixtures that do not have `scope="function"`.

Comment (Jaybsoni) [reply tests/test_device.py:170]: Ahh ok! Sorry I misunderstood your initial comment. If its not complaining then we should be fine 👍🏼

File: tests/test_hermitian_edge_cases.py [hunk 7]
```diff
@@ -1,23 +1,36 @@
...
-VARPHI = np.linspace(0.02, 1, 3)
 
 
 @pytest.mark.parametrize("shots", [None, 1000000])
-@pytest.mark.parametrize("theta,phi,varphi", list(zip(THETA, PHI, VARPHI)))
 class TestEdgeHermitian:
-    def test_hermitian_two_wires_identity_expectation_only_hermitian(
-        self, shots, theta, phi, varphi
-    ):
+    """Test Hermitian edge cases."""
```
[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/test_hermitian_edge_cases.py:30]: They were not all using all parameters, so I moved the paramerization inside :)

**Pull request merged**